scalafix-tests/unit/src/test/scala/scalafix/tests/core/PatchSuite.scala

```diff
@@ -93,4 +93,85 @@ class PatchSuite extends SyntacticRuleSuite {
       |object A
       |""".stripMargin
   )
+
+  val trimLeftIncludingNewLineRule: Rule = Rule.syntactic("trimLeft") { ctx =>
+    ctx.tokens.find(_.is[Token.KwVal]).map(ctx.trimLeft(_)).asPatch
+  }
+
+  check(
+    trimLeftIncludingNewLineRule,
+    "trimLeft removes spaces and newline before `val`",
+    original,
+    """// Foobar
+      |object a {val x = 2
```

Comment:
What happens if there is a // comment after the open curly brace? We should not remove newlines trailing such comments.

Reply:
We could test this by trimming left of "object"

Reply:
In the example above, the result would be:
```
// Foobar object a {
```
I put some conditions to prevent this issue (comments in the format `/* ... */` are OK). The same would occur when trimming on the right side of a comment though....

Comments are not the only case where trimming would cause an invalid output. Example:

```
object Foo {
   val a = 0
   val b = 0
}
```

`ctx.trimLeft(bValToken)`

Output:

```
object Foo {
   val a = 0 val b = 0
}
```

I feel that the current behavior for newlines is not very intuitive and useful in practice. I can't think of a valid scenario where I would like to get rid of the left/right newline and move the remaining code on a particular line (or line below) up. Is there any use case that comes to your mind? From the example given above and the use case @gabro described in #370, maybe the right API would be sth like `ctx.removeLine(token)` (`trim spaces` is still useful). What do you think?